features/306.ngsild.BigDataSpark.feature

```diff
@@ -0,0 +1,132 @@
+Feature: test tutorial 306 NGSI-LD Big Data Analysis (Spark)
+
+  This is the feature file of the FIWARE Step by Step tutorial Big Data Analysis with Spark- NGSI-LD
+
+  url: https://ngsi-ld-tutorials.readthedocs.io/en/latest/big-data-spark.html
+  git-clone: https://github.com/FIWARE/tutorials.Big-Data-Spark.git
+  git-directory: /tmp/tutorials.Big-Data-Spark
+  shell-commands: git checkout NGSI-LD ; mkdir cosmos-examples ; ./services create  ; ./services start
+  clean-shell-commands: ./services stop ; docker kill  spark-worker-1 ; docker kill spark-master ; docker rm  spark-worker-1 ; docker rm spark-master
+
+  Background:
+    Given I set the tutorial 306 NGSI-LD - Big Data analysis with Spark
+
+    # COMPILING A JAR FILE FOR SPARK --
+    Scenario: Compiling a Jar File For Spark
+        Given I prepare the script create-spark-jar.sh to be run
+        And   I set the environ variable WORKING_DIR to "cosmos-examples" under git
+        And   I set the environ variable OUTPUT_DIR to "cosmos-examples/target" under git
+        When  I run the script as in the tutorial page
+        Then  I expect the scripts shows a result of 0
+
+    #   LOGGER - READING CONTEXT DATA STREAMS
+    # - Logger - Installing the JAR
+    Scenario: Run the generated jar in Spark. This process won't end
+        Given I prepare the script run-spark-ld-jar.sh to be run
+        When  I open a new shell terminal spark-run-jar and run "features/data/306.ngsild.Big_Data_Spark/run-spark-ld-jar.sh"
+        And   I wait "2" seconds
+        Then  everything is ok
+
+    Scenario Outline: Registering spark logger
+        When I prepare a POST HTTP request for "<description>" to "http://localhost:1026/ngsi-ld/v1/subscriptions/"
+        And  I set header Content-Type to application/ld+json
+        And  I set header NGSILD-Tenant to openiot
+        And  I set the body request as described in <file>
+        And  I perform the request
+        Then  I receive a HTTP "201" response code
+        Examples:
+            | description                | file            |
+            | notify of animal locations | 01.request.json |
+
+    # Request 2 -
+  Scenario: Check the subscriptions for Spark Logger to ngsi-ld
+    When  I send GET HTTP request to "http://localhost:1026/ngsi-ld/v1/subscriptions/"
+    And   I set header NGSILD-Tenant to openiot
+    Then  I receive a HTTP "200" response code
+
+    # Populate things....
+     # Somehow I need to add data to Sensors so they can add data to Cratedb
+    # This is not in tutorial, but it is the way to add data without Web dashboard
+    ## Example post: 12:53:47 PM HTTP POST http://iot-agent:7896/iot/d?i=filling001&k=854782081 f|0.45
+    Scenario Outline: Communicating with IoT Devices: Using Actuators
```

Comment:
You are not using actuators... I suggest to change the title to something like:

Sending some mockup data to the IoT Device

Reply:
LGTM